core/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestGetSnapshotsAction.java

```diff
@@ -52,10 +60,19 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         String repository = request.param("repository");
         String[] snapshots = request.paramAsStringArray("snapshot", Strings.EMPTY_ARRAY);
 
-        GetSnapshotsRequest getSnapshotsRequest = getSnapshotsRequest(repository).snapshots(snapshots);
+        final GetSnapshotsRequest getSnapshotsRequest = getSnapshotsRequest(repository).snapshots(snapshots);
         getSnapshotsRequest.ignoreUnavailable(request.paramAsBoolean("ignore_unavailable", getSnapshotsRequest.ignoreUnavailable()));
         getSnapshotsRequest.verbose(request.paramAsBoolean("verbose", getSnapshotsRequest.verbose()));
         getSnapshotsRequest.masterNodeTimeout(request.paramAsTime("master_timeout", getSnapshotsRequest.masterNodeTimeout()));
-        return channel -> client.admin().cluster().getSnapshots(getSnapshotsRequest, new RestToXContentListener<>(channel));
+        return channel -> client.admin().cluster().getSnapshots(getSnapshotsRequest,
+            new RestBuilderListener<GetSnapshotsResponse>(channel) {
+                @Override
+                public RestResponse buildResponse(GetSnapshotsResponse response, XContentBuilder builder) throws Exception {
+                    Map<String, String> params = Collections.singletonMap("verbose", Boolean.toString(getSnapshotsRequest.verbose()));
```

Comment:
I don't think is necessary `request` is passed as`params`. So, "verbose" is already in `params` if it was specified, you just need to be careful resolving defaults.